pkgs/applications/graphics/structorizer/default.nix

```diff
@@ -0,0 +1,84 @@
+{ stdenv, lib, fetchFromGitHub, jdk11, makeDesktopItem, makeWrapper
+, copyDesktopItems }:
+
+stdenv.mkDerivation rec {
+  pname = "structorizer";
+  version = "3.32-11";
+
+  desktopItems = [
+    (makeDesktopItem {
+      type = "Application";
+      name = "Structorizer";
+      desktopName = "Structorizer";
+      genericName = "Diagram creator";
+      comment = meta.description;
+      icon = "structorizer";
+      exec = "structorizer";
+      terminal = false;
+      categories = [
+        "Development"
+        "Graphics"
+        "VectorGraphics"
+        "RasterGraphics"
+        "ComputerScience"
+      ];
+      keywords = [ "nsd" "diagrams" ];
+    })
+  ];
+
+  src = fetchFromGitHub {
+    owner = "fesch";
+    repo = "Structorizer.Desktop";
+    rev = version;
+    hash = "sha256-rGyeOcGm6uBplgTjMIOy/xRekfHacwDy9kkMigmRSdk=";
+  };
+
+  nativeBuildInputs = [ jdk11 makeWrapper copyDesktopItems ];
+
+  patches = [ ./makeStructorizer.patch ./makeBigJar.patch ];
```

Comment:
you could've done this in one patch, but imo you can leave it like that, since there is nothing speaking against separating it either.

Reply:
I'd rather leave it.